velox/connectors/hive/iceberg/tests/IcebergReadTest.cpp

```diff
@@ -327,37 +335,44 @@ class HiveIcebergTest : public HiveConnectorTestBase {
     for (int j = 0; j < vectorSizes.size(); j++) {
       auto data = makeContinuousIncreasingValues(
           startingValue, startingValue + vectorSizes[j]);
-      VectorPtr c0 = vectorMaker_.flatVector<int64_t>(data);
+      VectorPtr c0 = makeFlatVector<int64_t>(data);
       vectors.push_back(makeRowVector({"c0"}, {c0}));
       startingValue += vectorSizes[j];
     }
 
     return vectors;
   }
 
-  std::shared_ptr<ConnectorSplit> makeIcebergSplit(
+  std::vector<std::shared_ptr<ConnectorSplit>> makeIcebergSplits(
       const std::string& dataFilePath,
-      const std::vector<IcebergDeleteFile>& deleteFiles = {}) {
+      const std::vector<IcebergDeleteFile>& deleteFiles = {},
+      const uint32_t splitCount = 1) {
     std::unordered_map<std::string, std::optional<std::string>> partitionKeys;
     std::unordered_map<std::string, std::string> customSplitInfo;
     customSplitInfo["table_format"] = "hive-iceberg";
 
     auto file = filesystems::getFileSystem(dataFilePath, nullptr)
                     ->openFileForRead(dataFilePath);
     const int64_t fileSize = file->size();
+    std::vector<std::shared_ptr<ConnectorSplit>> splits;
+    const uint64_t splitSize = std::ceil((fileSize) / splitCount);
```

Comment:
If we use ceil, the total sizes might exceed the file size. Either use floor or guard the splitSize on line 366 to be <= remaining file size